Nomad 0.6.0 arm32v7: panic

### Nomad version
Nomad v0.6.0

### Operating system and Environment details
Ubuntu 16.04.3 LTS

### Issue
Panic

### Reproduction steps
Download nomad_0.6.0_linux_arm.zip, bootstrap server:
`nomad agent -server -data-dir /var/lib/nomad -bootstrap-expect 1 &`

### Nomad Server logs (if appropriate)
```
==> WARNING: Bootstrap mode enabled! Potentially unsafe operation.
    No configuration files loaded
==> Starting Nomad agent...
==> Nomad agent configuration:

                Client: false
             Log Level: INFO
                Region: global (DC: dc1)
                Server: true
               Version: 0.6.0

==> Nomad agent started! Log data will stream in below:

    2017/08/04 11:03:12 [INFO] raft: Initial configuration (index=1): [{Suffrage:Voter ID:169.254.155.20:4647 Address:169.254.155.20:4647}]
    2017/08/04 11:03:12 [INFO] raft: Node at 169.254.155.20:4647 [Follower] entering Follower state (Leader: "")
    2017/08/04 11:03:13 [INFO] serf: EventMemberJoin: arm32v7-host.global 169.254.155.20
    2017/08/04 11:03:13.018242 [INFO] nomad: starting 4 scheduling worker(s) for [service batch system _core]
    2017/08/04 11:03:13.019806 [INFO] nomad: adding server arm32v7-host.global (Addr: 169.254.155.20:4647) (DC: dc1)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x4 pc=0x11e60]

goroutine 27 [running]:
sync/atomic.storeUint64(0x11982774, 0x1, 0x0)
/opt/gopath/src/github.com/hashicorp/nomad/command/agent/consul/client.go:153 +0x34
github.com/hashicorp/nomad/command/agent/consul.(*ServiceClient).sync(0x11982720, 0x1161c4c0, 0x11624f6c)
/opt/gopath/src/github.com/hashicorp/nomad/command/agent/consul/client.go:177 +0x108
created by github.com/hashicorp/nomad/command/agent.(*Agent).setupConsul
/opt/gopath/src/github.com/hashicorp/nomad/
```
## Extra info

- Clean testing environment, no prior nomad installation, no nomad clients.
- Issue does not happen without a running Consul.
- `/opt/go/src/sync/atomic/64bit_arm.go:20 +0x3c`, test instance is arm 32bit.

## Bonus Error
`nomad agent -server -data-dir /var/lib/nomad -bootstrap-expect 1 -name foo 2>&1 |head -3
`

--- a/command/agent/consul/client.go
+++ b/command/agent/consul/client.go
@@ -119,7 +119,7 @@ type ServiceClient struct {
 
 	// seen is 1 if Consul has ever been seen; otherise 0. Accessed with
 	// atomics.
-	seen int64
+	seen int32
 }
 
 // NewServiceClient creates a new Consul ServiceClient from an existing Consul API
@@ -150,13 +150,13 @@ const seen = 1
 // markSeen marks Consul as having been seen (meaning at least one operation
 // has succeeded).
 func (c *ServiceClient) markSeen() {
-	atomic.StoreInt64(&c.seen, seen)
+	atomic.StoreInt32(&c.seen, seen)
 }
 
 // hasSeen returns true if any Consul operation has ever succeeded. Useful to
 // squelch errors if Consul isn't running.
 func (c *ServiceClient) hasSeen() bool {
-	return atomic.LoadInt64(&c.seen) == seen
+	return atomic.LoadInt32(&c.seen) == seen
 }
 
 // Run the Consul main loop which retries operations against Consul. It should

